test/integration/plugin/tests/test-plugin-resource-launch.hpp

```diff
@@ -0,0 +1,89 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing basic integration tests for plugins with launch.
+///
+
+#ifndef __TEST_PLUGIN_RESOURCE_LAUNCH_HPP__
+#define __TEST_PLUGIN_RESOURCE_LAUNCH_HPP__
+
+#include "test-plugin.hpp"
+
+
+// Check that the plugin is called with the right Platform.
+// Check that the plugin is called the correct number of times,
+// once before and after each launch capture for the capture counter,
+// once before and after each launch invocation for the launch counter.
+
+// test with basic launch
+template <typename LaunchPolicy,
+          typename WORKING_RES,
+          RAJA::Platform PLATFORM>
+void PluginResourceLaunchTestImpl()
+{
+  SetupPluginVars spv(WORKING_RES::get_default());
+
+  CounterData* data = plugin_test_resource->allocate<CounterData>(10);
+
+  for (int i = 0; i < 10; i++) {
+
+    //Keep PluginTestCallable within a scope to ensure
+    //destruction, consistent with other test
+    {
+      PluginTestCallable p_callable{data};
+
+      RAJA::launch<LaunchPolicy>
+        (WORKING_RES::get_default(), RAJA::LaunchParams(RAJA::Teams(1), RAJA::Threads(1)),
```

Comment:
Is this okay? or do we want to try the non-default option?

Reply:
I think testing all cases that you think users may want is what we should do.

Reply:
I don't think which value of the resource matters as long as there is a test that exercises the resource code path.